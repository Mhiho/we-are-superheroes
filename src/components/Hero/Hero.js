import React from 'react'


const hero = props => {

    return (
        <div onClick={props.click}>
            <h1>{props.name}</h1>
            <img alt="heroImage" src={props.src}/>
        </div>
    )
}

export default hero;