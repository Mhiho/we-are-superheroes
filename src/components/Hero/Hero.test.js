import React from 'react';
import renderer from 'react-test-renderer';

import Hero from './Hero';

it('renders correctly when there is no props', () => {
  const tree = renderer.create(<Hero />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly when there is src', () => {
    const src = 'path';
    const tree = renderer.create(<Hero src={src} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly when there is name', () => {
    const name = 'string';
    const tree = renderer.create(<Hero name={name} />).toJSON();
    expect(tree).toMatchSnapshot();
  });