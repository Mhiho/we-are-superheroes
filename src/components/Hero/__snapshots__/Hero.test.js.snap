// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is name 1`] = `
<li
  className="list-group-item"
>
  <img
    alt="heroImage"
    className="rounded float-left"
    style={
      Object {
        "width": "100px",
      }
    }
  />
  <div
    className="h-100 d-flex align-items-center justify-content-center"
  >
    <h3>
      string
    </h3>
  </div>
</li>
`;

exports[`renders correctly when there is no props 1`] = `
<li
  className="list-group-item"
>
  <img
    alt="heroImage"
    className="rounded float-left"
    style={
      Object {
        "width": "100px",
      }
    }
  />
  <div
    className="h-100 d-flex align-items-center justify-content-center"
  >
    <h3 />
  </div>
</li>
`;

exports[`renders correctly when there is src 1`] = `
<li
  className="list-group-item"
>
  <img
    alt="heroImage"
    className="rounded float-left"
    src="path"
    style={
      Object {
        "width": "100px",
      }
    }
  />
  <div
    className="h-100 d-flex align-items-center justify-content-center"
  >
    <h3 />
  </div>
</li>
`;
